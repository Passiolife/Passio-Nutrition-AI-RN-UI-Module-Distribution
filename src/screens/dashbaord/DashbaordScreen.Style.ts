import type { TextStyle, ViewStyle } from 'react-native';
import { Dimensions } from 'react-native';
import { palette } from '../../components/palette';
import {
  baseNormalTextStyle,
  baseSmallTextStyle,
  baseTitleNormalTextStyle,
} from '../Screen.Style';

export const baseContainer: ViewStyle = {
  flex: 1,
  backgroundColor: palette.white,
  paddingHorizontal: 24,
};

export const safeAreaContainer: ViewStyle = {
  flex: 1,
  backgroundColor: palette.white,
};

export const dashboardBaseContainer: ViewStyle = {
  flex: 1,
  backgroundColor: palette.white,
  marginTop: 30,
};

// Start Tracking Card View
export const startNutritionTrackingCardContainer: ViewStyle = {
  alignSelf: 'center',
  marginHorizontal: 8,
  elevation: 5,
  shadowColor: 'black',
  shadowOffset: { width: 0, height: 2 },
  shadowRadius: 10,
  marginTop: 16,
  paddingVertical: 24,
  justifyContent: 'space-between',
  shadowOpacity: 0.1,
  padding: 5,
  backgroundColor: palette.violetRed,
  borderRadius: 16,
  overflow: 'hidden',
};

export const nutritionCardTitleStyle: TextStyle = {
  ...baseTitleNormalTextStyle,
  paddingStart: 16,
  fontSize: 20,
  minHeight: 100,
  paddingEnd: Dimensions.get('window').width / 3,
  color: palette.white,
};

export const startTrackingStyle: TextStyle = {
  fontSize: 16,
  fontWeight: '600',
  borderWidth: 1.5,
  marginHorizontal: 16,
  marginTop: 24,
  width: 150,
  textAlign: 'center',
  paddingVertical: 8,
  borderColor: palette.white,
  borderRadius: 16,
  color: palette.white,
};

export const nutritionTrackContainer: ViewStyle = {
  width: Dimensions.get('window').width - 34,
  alignSelf: 'center',
  height: 'auto',
  elevation: 15,
  shadowColor: 'black',
  marginTop: 16,
  paddingVertical: 24,
  shadowOffset: { width: 0, height: 2 },
  shadowRadius: 10,
  justifyContent: 'space-between',
  shadowOpacity: 0.1,
  padding: 5,
  backgroundColor: palette.white,
  borderRadius: 18,
};

export const nutritionTrackTextStyle: TextStyle = {
  ...baseTitleNormalTextStyle,
  paddingStart: 16,
  fontSize: 16,
  paddingEnd: Dimensions.get('window').width / 3,
  color: palette.charcoal,
};

export const nutritionTrackInfoTextStyle: TextStyle = {
  ...baseSmallTextStyle,
  paddingStart: 16,
  fontSize: 13,
  paddingTop: 4,
  color: palette.charcoal,
};

export const buttonTrackNutrition: TextStyle = {
  ...baseNormalTextStyle,
  borderWidth: 1,
  marginHorizontal: 8,
  textAlign: 'center',
  paddingVertical: 6,
  alignSelf: 'center',
  fontWeight: '500',
  paddingHorizontal: 16,
  borderRadius: 16,
  fontSize: 15,
  color: palette.black,
};
