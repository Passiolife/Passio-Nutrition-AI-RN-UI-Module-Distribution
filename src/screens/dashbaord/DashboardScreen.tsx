import React from 'react';
import { SafeAreaView } from 'react-native';
import { safeAreaContainer } from './DashbaordScreen.Style';

export const DashboardScreen = () => {
  return <SafeAreaView style={safeAreaContainer} />;
};
