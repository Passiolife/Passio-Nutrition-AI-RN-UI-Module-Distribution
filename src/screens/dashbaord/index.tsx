export * from './DashboardScreen';
export * from './DashbaordScreen.Style';
export * from './utils';
