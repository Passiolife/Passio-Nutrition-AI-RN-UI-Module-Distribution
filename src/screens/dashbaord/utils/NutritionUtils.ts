import type { FoodLog } from '../../../models';
import { totalAmountOfNutrient } from '../../editFoodLogs';
import type { NutritionProfile } from '../../../models';

export interface DailyMacroNutrientAndCalories {
  amountOfCalories: number;
  amountOfCarbs: number;
  amountOfProtein: number;
  amountOfFat: number;
}
export const calculateDailyMacroNutrition = (
  logs: FoodLog[]
): DailyMacroNutrientAndCalories => {
  let totalCalories = 0;
  let totalCarbs = 0;
  let totalProtein = 0;
  let totalFat = 0;

  logs.forEach((value) => {
    totalCalories =
      totalCalories + totalAmountOfNutrient(value.foodItems, 'calories');
    totalCarbs = totalCarbs + totalAmountOfNutrient(value.foodItems, 'carbs');
    totalProtein =
      totalProtein + totalAmountOfNutrient(value.foodItems, 'protein');
    totalFat = totalFat + totalAmountOfNutrient(value.foodItems, 'fat');
  });
  return {
    amountOfCalories: totalCalories,
    amountOfCarbs: totalCarbs,
    amountOfProtein: totalProtein,
    amountOfFat: totalFat,
  };
};

export interface NutritionTarget {
  targetCalories: number;
  targetCarbs: number;
  targetProtein: number;
  targetFat: number;
}

export const calculateTargetNutritionProfile = (
  profile: NutritionProfile | undefined
): NutritionTarget => {
  let targetCalories = 0;
  let targetCarbs = 0;
  let targetProtein = 0;
  let targetFat = 0;

  if (profile !== undefined) {
    targetCalories = profile.caloriesTarget;
    targetCarbs = (targetCalories * profile.carbsPercentage) / 100 / 4;
    targetFat = (targetCalories * profile.fatPercentage) / 100 / 9;
    targetProtein = (targetCalories * profile.proteinPercentage) / 100 / 4;
  }

  return {
    targetCalories: targetCalories,
    targetCarbs: Math.round(targetCarbs),
    targetProtein: Math.round(targetProtein),
    targetFat: Math.round(targetFat),
  };
};

export function caloriesText(foodLog: FoodLog, text: string = 'g'): string {
  let calorieText = '';
  let calorie: number | undefined = totalAmountOfNutrient(
    foodLog.foodItems,
    'calories'
  );
  if (calorie !== undefined) {
    calorieText = `${calorie.toFixed(0)} ${text}`;
  }
  return calorieText;
}
