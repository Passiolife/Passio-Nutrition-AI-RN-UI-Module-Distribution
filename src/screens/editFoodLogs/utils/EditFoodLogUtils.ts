import type {
  FoodItem,
  NutrientType,
  ServingUnit,
  Nutrient,
  ServingInfo,
} from '../../../models';
import type { ComputedWeight } from '../../../models/ComputedWeight';

export function maxSliderValue(qty: number): number {
  const sliderMultiplier = 5.0;
  if (qty <= 0) {
    qty = 1;
  }
  if (qty > 100 && qty <= 500) {
    return 500;
  } else if (qty > 0 && qty <= 5) {
    return 5;
  }
  return sliderMultiplier * qty;
}

export function steps(sliderMaxValue: number): number {
  if (sliderMaxValue >= 0 && sliderMaxValue <= 10) {
    return 0.5;
  } else if (sliderMaxValue >= 10 && sliderMaxValue <= 100) {
    return 1;
  } else {
    return 10;
  }
}

export function totalAmountOfNutrient(
  foodItems: FoodItem[],
  nutrientType: NutrientType
): number {
  const totalAmount = totalAmountOfNutrientWithoutRound(
    foodItems,
    nutrientType
  );
  return totalAmount > 1
    ? Math.floor(totalAmount)
    : parseFloat(totalAmount.toFixed(1));
}
export function totalAmountOfNutrientWithoutRound(
  foodItems: FoodItem[],
  nutrientType: NutrientType
): number {
  let totalAmount = 0;
  foodItems.forEach((foodItem) => {
    foodItem.nutrients.forEach((value) => {
      if (value.id === nutrientType) {
        totalAmount = totalAmount + value.amount;
      }
    });
  });
  return totalAmount;
}

export function calculateMassOfServingUnit(
  servingUnits: ServingUnit[],
  selectedUnit: string
): number {
  let unit = 1;
  servingUnits?.forEach((value) => {
    if (value.unit === selectedUnit) {
      unit = value.mass;
    }
  });
  return unit;
}

export function calculateComputedWeightAmount(
  qty: number,
  servingUnits: ServingUnit[],
  unit: string
) {
  const result = qty * calculateMassOfServingUnit(servingUnits, unit);
  return result < 10 ? result : Math.ceil(result);
}

export function updatingFoodItem(
  servingInfo: ServingInfo,
  foodItems: FoodItem[],
  updatedQty: number,
  updateUnit: string
): FoodItem[] {
  let originWeight = calculateComputedWeightAmount(
    servingInfo.selectedQuantity,
    servingInfo.servingUnits,
    servingInfo.selectedUnit
  );
  if (originWeight === 0) {
    originWeight = 1;
  }
  let newWeight = calculateComputedWeightAmount(
    updatedQty,
    servingInfo.servingUnits,
    updateUnit
  );

  if (newWeight === 0) {
    newWeight = 1;
  }

  let copyFoodItems: FoodItem[] = [];
  foodItems.forEach((foodItem) => {
    let nutrients: Nutrient[] = [];
    foodItem.nutrients.forEach((nutrient: Nutrient) => {
      const amount = (newWeight * nutrient.amount) / originWeight;
      const copyOfNutrient: Nutrient = {
        ...nutrient,
        ...{ amount: amount },
      };
      nutrients.push(copyOfNutrient);
    });
    const copyOfComputedWeight: ComputedWeight = {
      ...foodItem.computedWeight,
      value: newWeight,
    };
    const newFoodItem: FoodItem = {
      ...foodItem,
      selectedQuantity: parseFloat(updatedQty.toString()),
      selectedUnit: updateUnit,
      nutrients: nutrients,
      computedWeight: copyOfComputedWeight,
    };
    copyFoodItems.push(newFoodItem);
  });
  return copyFoodItems;
}
