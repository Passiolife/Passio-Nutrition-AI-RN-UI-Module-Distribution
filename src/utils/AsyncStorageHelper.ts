import AsyncStorage from '@react-native-async-storage/async-storage';
import type { ActionPlanType } from '../models';
import type { AnalyticsFoodLogs } from '../models/PassioAnalytics';

const storeData = async (key: string, value: any) => {
  return new Promise((resolve, reject) => {
    try {
      resolve(AsyncStorage.setItem(key, value));
    } catch (e) {
      reject(e);
    }
  });
};

export class AsyncStorageHelper {
  static async checkOnBoardingCompleted() {
    try {
      const value = await AsyncStorage.getItem('hasCompletedOnboarding');
      return value !== null;
    } catch (e) {
      return false;
    }
  }

  static setOnBoardingCompleted() {
    return storeData('hasCompletedOnboarding', '1');
  }

  static setAnalyticsFoodLogs(analyticsFoodLogs: AnalyticsFoodLogs[]) {
    return storeData('analyticsFoodLogs', JSON.stringify(analyticsFoodLogs));
  }

  static async getAnalyticsFoodLogs() {
    const value = await AsyncStorage.getItem('analyticsFoodLogs');
    return value ? (JSON.parse(value) as AnalyticsFoodLogs[]) : [];
  }

  static setLoggedInDate() {
    return storeData('loggedInDate', new Date().toISOString());
  }

  static async getLoggedInDate() {
    try {
      const value = await AsyncStorage.getItem('loggedInDate');
      return value !== null ? new Date(value) : null;
    } catch (e) {
      return null;
    }
  }

  static setActionPlan(actionPlan: ActionPlanType | null) {
    return storeData('actionPlan', actionPlan);
  }

  static async getActionPlan(): Promise<ActionPlanType | null> {
    try {
      const value = await AsyncStorage.getItem('actionPlan');
      return value as ActionPlanType | null;
    } catch (e) {
      return null;
    }
  }
}
