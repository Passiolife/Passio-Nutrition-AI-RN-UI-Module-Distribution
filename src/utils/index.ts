import { dateFormatter, timeFormatter } from './DateFormatter';
export { dateFormatter, timeFormatter };
import {
  isToday,
  currentTimeStamp,
  getOneWeek,
  getOneMonth,
  getTwoWeek,
  getDateWeekNumber,
  getWeekDatesUsingWeekNumber,
} from './DateUtils';
export * from './withLoading';
export * from './Dimensions';
export * from './Font';
export * from './toast';
export * from './MealPlanUtils';
export * from './PassioUtils';
export * from './QuickResultUtils';
export * from './ScaningUtils';
export * from './quickSuggestionUtils';
export * from './passioFoodDataInfoUtils';
export {
  isToday,
  currentTimeStamp,
  getOneWeek,
  getTwoWeek,
  getOneMonth,
  getDateWeekNumber,
  getWeekDatesUsingWeekNumber,
};
