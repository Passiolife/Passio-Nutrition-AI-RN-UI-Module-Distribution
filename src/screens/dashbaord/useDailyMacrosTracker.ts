import { useServices } from '../../contexts';
import { resourceCache, useAsyncResource } from 'use-async-resource';
import { getDailyMacrosProgress } from '../../utils/DataServiceHelper';
import { useEffect } from 'react';
import { useIsFocused } from '@react-navigation/native';

export function useDailyMacrosTracker(startDate: Date, endDate: Date) {
  const services = useServices();
  const isFocused = useIsFocused();

  const [nutritionTracker] = useAsyncResource(
    getDailyMacrosProgress,
    startDate,
    endDate,
    services
  );
  const dailyTracker = nutritionTracker();

  useEffect(() => {
    resourceCache(getDailyMacrosProgress).clear();
  }, [isFocused]);

  return {
    dailyTracker,
  };
}
