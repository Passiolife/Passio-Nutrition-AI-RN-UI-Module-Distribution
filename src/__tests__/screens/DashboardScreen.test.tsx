// @ts-nocheck
import React from 'react';
import { DashboardScreen } from '../../screens/dashbaord/DashboardScreen';
import { MockProviders } from '../provider/MockProviders';
import {
  render,
  type RenderAPI,
  waitForElementToBeRemoved,
} from '@testing-library/react-native';
import { AsyncStorageHelper } from '../../utils/AsyncStorageHelper';

const mockedDispatch = jest.fn();
jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: mockedDispatch,
    }),
    useIsFocused: () => jest.fn(),
  };
});

jest.mock('../../screens/scanning/usePassioAuthConfig', () => {
  return {
    usePassioAuthConfig: () => ({
      isReady: true,
    }),
  };
});

describe('Dashboard Screen :', () => {
  it('checked if user has not completed onBoarding and render screen ', async () => {
    let tree: RenderAPI;
    tree = await render(
      <>
        <MockProviders>
          <DashboardScreen />
        </MockProviders>
      </>
    );
    await waitForElementToBeRemoved(() => tree.queryByTestId('tesViewBlank'));
    expect(tree.queryByTestId('testViewOnBoardingCard')).not.toEqual(null);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('checked if user has completed onBoarding and render screen', async () => {
    let tree: RenderAPI;
    await AsyncStorageHelper.setOnBoardingCompleted();
    tree = await render(
      <>
        <MockProviders>
          <DashboardScreen />
        </MockProviders>
      </>
    );
    await waitForElementToBeRemoved(() => tree.queryByTestId('tesViewBlank'));
    expect(tree.queryByTestId('testViewNutritionDataCard')).not.toEqual(null);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
