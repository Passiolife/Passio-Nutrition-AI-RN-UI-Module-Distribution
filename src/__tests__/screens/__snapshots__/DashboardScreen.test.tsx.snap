// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Screen : checked if user has completed onBoarding and render screen 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View>
    <View
      style={
        {
          "alignSelf": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 18,
          "elevation": 15,
          "height": "auto",
          "justifyContent": "space-between",
          "marginTop": 16,
          "padding": 5,
          "paddingVertical": 24,
          "shadowColor": "black",
          "shadowOffset": {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 10,
          "width": 716,
        }
      }
      testID="testViewNutritionDataCard"
    >
      <View>
        <Text
          style={
            {
              "color": "#333333",
              "fontSize": 16,
              "fontWeight": "normal",
              "paddingEnd": 250,
              "paddingStart": 16,
            }
          }
        >
          Nutrition Tracker
        </Text>
        <Text
          style={
            {
              "color": "#333333",
              "fontSize": 13,
              "fontWeight": "400",
              "paddingStart": 16,
              "paddingTop": 4,
            }
          }
        >
          Today's macronutrients & calories
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginHorizontal": 16,
            "marginVertical": 24,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
            }
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                {
                  "height": 142.85714285714286,
                  "width": 142.85714285714286,
                },
              ]
            }
          >
            <View
              style={
                [
                  {
                    "position": "absolute",
                  },
                  {
                    "borderBottomWidth": 3.5,
                    "borderLeftWidth": 3.5,
                    "borderRadius": 71.42857142857143,
                    "borderRightWidth": 3.5,
                    "borderTopWidth": 3.5,
                    "height": 142.85714285714286,
                    "width": 142.85714285714286,
                  },
                  {
                    "borderColor": "#e4e4e4",
                    "borderWidth": 3.5,
                  },
                ]
              }
            />
            <View
              style={
                [
                  {
                    "borderBottomColor": "transparent",
                    "borderLeftColor": "transparent",
                    "position": "absolute",
                  },
                  {
                    "transform": [
                      {
                        "rotateZ": "-135deg",
                      },
                    ],
                  },
                  {
                    "borderBottomWidth": 3.5,
                    "borderLeftWidth": 3.5,
                    "borderRadius": 71.42857142857143,
                    "borderRightWidth": 3.5,
                    "borderTopWidth": 3.5,
                    "height": 142.85714285714286,
                    "width": 142.85714285714286,
                  },
                  {
                    "borderRightColor": "#f2bc34",
                    "borderTopColor": "#f2bc34",
                  },
                  {
                    "borderRightWidth": 3.5,
                    "borderTopWidth": 3.5,
                  },
                ]
              }
            />
            <View
              style={
                [
                  {
                    "borderBottomColor": "transparent",
                    "borderLeftColor": "transparent",
                    "position": "absolute",
                  },
                  {
                    "borderBottomWidth": 3.5,
                    "borderLeftWidth": 3.5,
                    "borderRadius": 71.42857142857143,
                    "borderRightWidth": 3.5,
                    "borderTopWidth": 3.5,
                    "height": 142.85714285714286,
                    "width": 142.85714285714286,
                  },
                  {
                    "borderRightColor": "#e4e4e4",
                    "borderTopColor": "#e4e4e4",
                  },
                  {
                    "transform": [
                      {
                        "rotateZ": "-135deg",
                      },
                    ],
                  },
                ]
              }
            />
            <View
              style={
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#333333",
                    "fontSize": 13,
                    "fontWeight": "600",
                    "textAlign": "center",
                  }
                }
              >
                0
              </Text>
              <View
                style={
                  {
                    "borderBottomColor": "#333333",
                    "borderBottomWidth": 0.5,
                    "width": 45,
                  }
                }
              />
              <Text
                style={
                  {
                    "color": "#333333",
                    "fontSize": 13,
                    "fontWeight": "400",
                    "paddingHorizontal": 8,
                    "textAlign": "center",
                  }
                }
              >
                2100
                 
                cal
              </Text>
            </View>
          </View>
          <Text
            style={
              {
                "color": "#333333",
                "fontSize": 13,
                "fontWeight": "400",
                "marginTop": 5,
                "textTransform": "capitalize",
              }
            }
          >
            Calories
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
            }
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                {
                  "height": 142.85714285714286,
                  "width": 142.85714285714286,
                },
              ]
            }
          >
            <View
              style={
                [
                  {
                    "position": "absolute",
                  },
                  {
                    "borderBottomWidth": 3.5,
                    "borderLeftWidth": 3.5,
                    "borderRadius": 71.42857142857143,
                    "borderRightWidth": 3.5,
                    "borderTopWidth": 3.5,
                    "height": 142.85714285714286,
                    "width": 142.85714285714286,
                  },
                  {
                    "borderColor": "#e4e4e4",
                    "borderWidth": 3.5,
                  },
                ]
              }
            />
            <View
              style={
                [
                  {
                    "borderBottomColor": "transparent",
                    "borderLeftColor": "transparent",
                    "position": "absolute",
                  },
                  {
                    "transform": [
                      {
                        "rotateZ": "-135deg",
                      },
                    ],
                  },
                  {
                    "borderBottomWidth": 3.5,
                    "borderLeftWidth": 3.5,
                    "borderRadius": 71.42857142857143,
                    "borderRightWidth": 3.5,
                    "borderTopWidth": 3.5,
                    "height": 142.85714285714286,
                    "width": 142.85714285714286,
                  },
                  {
                    "borderRightColor": "#4b99ff",
                    "borderTopColor": "#4b99ff",
                  },
                  {
                    "borderRightWidth": 3.5,
                    "borderTopWidth": 3.5,
                  },
                ]
              }
            />
            <View
              style={
                [
                  {
                    "borderBottomColor": "transparent",
                    "borderLeftColor": "transparent",
                    "position": "absolute",
                  },
                  {
                    "borderBottomWidth": 3.5,
                    "borderLeftWidth": 3.5,
                    "borderRadius": 71.42857142857143,
                    "borderRightWidth": 3.5,
                    "borderTopWidth": 3.5,
                    "height": 142.85714285714286,
                    "width": 142.85714285714286,
                  },
                  {
                    "borderRightColor": "#e4e4e4",
                    "borderTopColor": "#e4e4e4",
                  },
                  {
                    "transform": [
                      {
                        "rotateZ": "-135deg",
                      },
                    ],
                  },
                ]
              }
            />
            <View
              style={
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#333333",
                    "fontSize": 13,
                    "fontWeight": "600",
                    "textAlign": "center",
                  }
                }
              >
                0
              </Text>
              <View
                style={
                  {
                    "borderBottomColor": "#333333",
                    "borderBottomWidth": 0.5,
                    "width": 45,
                  }
                }
              />
              <Text
                style={
                  {
                    "color": "#333333",
                    "fontSize": 13,
                    "fontWeight": "400",
                    "paddingHorizontal": 8,
                    "textAlign": "center",
                  }
                }
              >
                289
                 
                g
              </Text>
            </View>
          </View>
          <Text
            style={
              {
                "color": "#333333",
                "fontSize": 13,
                "fontWeight": "400",
                "marginTop": 5,
                "textTransform": "capitalize",
              }
            }
          >
            Carbs
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
            }
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                {
                  "height": 142.85714285714286,
                  "width": 142.85714285714286,
                },
              ]
            }
          >
            <View
              style={
                [
                  {
                    "position": "absolute",
                  },
                  {
                    "borderBottomWidth": 3.5,
                    "borderLeftWidth": 3.5,
                    "borderRadius": 71.42857142857143,
                    "borderRightWidth": 3.5,
                    "borderTopWidth": 3.5,
                    "height": 142.85714285714286,
                    "width": 142.85714285714286,
                  },
                  {
                    "borderColor": "#e4e4e4",
                    "borderWidth": 3.5,
                  },
                ]
              }
            />
            <View
              style={
                [
                  {
                    "borderBottomColor": "transparent",
                    "borderLeftColor": "transparent",
                    "position": "absolute",
                  },
                  {
                    "transform": [
                      {
                        "rotateZ": "-135deg",
                      },
                    ],
                  },
                  {
                    "borderBottomWidth": 3.5,
                    "borderLeftWidth": 3.5,
                    "borderRadius": 71.42857142857143,
                    "borderRightWidth": 3.5,
                    "borderTopWidth": 3.5,
                    "height": 142.85714285714286,
                    "width": 142.85714285714286,
                  },
                  {
                    "borderRightColor": "#57d795",
                    "borderTopColor": "#57d795",
                  },
                  {
                    "borderRightWidth": 3.5,
                    "borderTopWidth": 3.5,
                  },
                ]
              }
            />
            <View
              style={
                [
                  {
                    "borderBottomColor": "transparent",
                    "borderLeftColor": "transparent",
                    "position": "absolute",
                  },
                  {
                    "borderBottomWidth": 3.5,
                    "borderLeftWidth": 3.5,
                    "borderRadius": 71.42857142857143,
                    "borderRightWidth": 3.5,
                    "borderTopWidth": 3.5,
                    "height": 142.85714285714286,
                    "width": 142.85714285714286,
                  },
                  {
                    "borderRightColor": "#e4e4e4",
                    "borderTopColor": "#e4e4e4",
                  },
                  {
                    "transform": [
                      {
                        "rotateZ": "-135deg",
                      },
                    ],
                  },
                ]
              }
            />
            <View
              style={
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#333333",
                    "fontSize": 13,
                    "fontWeight": "600",
                    "textAlign": "center",
                  }
                }
              >
                0
              </Text>
              <View
                style={
                  {
                    "borderBottomColor": "#333333",
                    "borderBottomWidth": 0.5,
                    "width": 45,
                  }
                }
              />
              <Text
                style={
                  {
                    "color": "#333333",
                    "fontSize": 13,
                    "fontWeight": "400",
                    "paddingHorizontal": 8,
                    "textAlign": "center",
                  }
                }
              >
                105
                 
                g
              </Text>
            </View>
          </View>
          <Text
            style={
              {
                "color": "#333333",
                "fontSize": 13,
                "fontWeight": "400",
                "marginTop": 5,
                "textTransform": "capitalize",
              }
            }
          >
            Protein
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
            }
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                {
                  "height": 142.85714285714286,
                  "width": 142.85714285714286,
                },
              ]
            }
          >
            <View
              style={
                [
                  {
                    "position": "absolute",
                  },
                  {
                    "borderBottomWidth": 3.5,
                    "borderLeftWidth": 3.5,
                    "borderRadius": 71.42857142857143,
                    "borderRightWidth": 3.5,
                    "borderTopWidth": 3.5,
                    "height": 142.85714285714286,
                    "width": 142.85714285714286,
                  },
                  {
                    "borderColor": "#e4e4e4",
                    "borderWidth": 3.5,
                  },
                ]
              }
            />
            <View
              style={
                [
                  {
                    "borderBottomColor": "transparent",
                    "borderLeftColor": "transparent",
                    "position": "absolute",
                  },
                  {
                    "transform": [
                      {
                        "rotateZ": "-135deg",
                      },
                    ],
                  },
                  {
                    "borderBottomWidth": 3.5,
                    "borderLeftWidth": 3.5,
                    "borderRadius": 71.42857142857143,
                    "borderRightWidth": 3.5,
                    "borderTopWidth": 3.5,
                    "height": 142.85714285714286,
                    "width": 142.85714285714286,
                  },
                  {
                    "borderRightColor": "#ee4b62",
                    "borderTopColor": "#ee4b62",
                  },
                  {
                    "borderRightWidth": 3.5,
                    "borderTopWidth": 3.5,
                  },
                ]
              }
            />
            <View
              style={
                [
                  {
                    "borderBottomColor": "transparent",
                    "borderLeftColor": "transparent",
                    "position": "absolute",
                  },
                  {
                    "borderBottomWidth": 3.5,
                    "borderLeftWidth": 3.5,
                    "borderRadius": 71.42857142857143,
                    "borderRightWidth": 3.5,
                    "borderTopWidth": 3.5,
                    "height": 142.85714285714286,
                    "width": 142.85714285714286,
                  },
                  {
                    "borderRightColor": "#e4e4e4",
                    "borderTopColor": "#e4e4e4",
                  },
                  {
                    "transform": [
                      {
                        "rotateZ": "-135deg",
                      },
                    ],
                  },
                ]
              }
            />
            <View
              style={
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#333333",
                    "fontSize": 13,
                    "fontWeight": "600",
                    "textAlign": "center",
                  }
                }
              >
                0
              </Text>
              <View
                style={
                  {
                    "borderBottomColor": "#333333",
                    "borderBottomWidth": 0.5,
                    "width": 45,
                  }
                }
              />
              <Text
                style={
                  {
                    "color": "#333333",
                    "fontSize": 13,
                    "fontWeight": "400",
                    "paddingHorizontal": 8,
                    "textAlign": "center",
                  }
                }
              >
                58
                 
                g
              </Text>
            </View>
          </View>
          <Text
            style={
              {
                "color": "#333333",
                "fontSize": 13,
                "fontWeight": "400",
                "marginTop": 5,
                "textTransform": "capitalize",
              }
            }
          >
            Fat
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            [
              {
                "alignSelf": "center",
                "borderRadius": 16,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 15,
                "fontWeight": "500",
                "marginHorizontal": 8,
                "paddingHorizontal": 16,
                "paddingVertical": 6,
                "textAlign": "center",
              },
              {
                "borderColor": "#123456",
                "color": "#123456",
              },
            ]
          }
        >
          Track Nutrition
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`Dashboard Screen : checked if user has not completed onBoarding and render screen  1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View>
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        {
          "alignSelf": "center",
          "backgroundColor": "#8F0C5C",
          "borderRadius": 16,
          "elevation": 5,
          "justifyContent": "space-between",
          "marginHorizontal": 8,
          "marginTop": 16,
          "overflow": "hidden",
          "padding": 5,
          "paddingVertical": 24,
          "shadowColor": "black",
          "shadowOffset": {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 10,
        }
      }
    >
      <Image
        resizeMode="cover"
        source={
          {
            "testUri": "../../../src/assets/bg_on_boarding_card.png",
          }
        }
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
        testID="testViewOnBoardingCard"
      />
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 20,
            "fontWeight": "normal",
            "minHeight": 100,
            "paddingEnd": 250,
            "paddingStart": 16,
          }
        }
      >
        Track Your Nutrition and Meet Your Goals
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            {
              "borderColor": "#FFFFFF",
              "borderRadius": 16,
              "borderWidth": 1.5,
              "color": "#FFFFFF",
              "fontSize": 16,
              "fontWeight": "600",
              "marginHorizontal": 16,
              "marginTop": 24,
              "paddingVertical": 8,
              "textAlign": "center",
              "width": 150,
            }
          }
        >
          Start Tracking
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
